### PR TITLE
debian: package systemd targets

### PR DESCRIPTION
**Pull request by tchaikov**
Fixes: http://tracker.ceph.com/issues/15573

### CODE REVIEW

**Comment by liewegas** [comment]
I don't think invoke-rc.d will do the right thing.  The 'start ceph-all' and 'start ceph-osd-all' etc lines are a bit of a hack so that the *wrapper* meta-service is running after a fresh install, when no daemons actually exist.  That way, when the daemon(s) is/are created, 'stop ceph-osd-all' will stop them.

It's been this way for a long time, and given that we're basic at the end of upstart anyway, I don't think we should change it..

**Comment by tchaikov** [comment]
changelog

- dropped the commit to s/start/invoke-rc.d/

File: debian/rules [hunk 1]
```diff
@@ -155,18 +154,28 @@ binary-arch: build install
...
 
 	install -d -m0755 debian/rbd-mirror/lib/systemd/system
 	install -m0644 systemd/ceph-rbd-mirror@.service debian/rbd-mirror/lib/systemd/system
 	sed -i s./etc/sysconfig/./etc/default/.g debian/rbd-mirror/lib/systemd/system/ceph-rbd-mirror@.service
+	install -m0644 systemd/ceph-rbd-mirror.target debian/rbd-mirror/lib/systemd/system
+
+	dh_systemd_enable
+	dh_installinit -p ceph-base --name ceph --no-start
+	dh_installinit -p radosgw --no-start
+	dh_systemd_start --restart-after-upgrade
```
Comment (liewegas) [inline debian/rules]: hmm, do we really want --restart-after-upgrade?  traditionally we haven't done that.

Comment (liewegas) [inline debian/rules]: i think we want --no-restart-on-upgrade

Comment (tchaikov) [inline debian/rules]: > --restart-after-upgrade
>           Do not stop the unit file until after the package upgrade has been completed.

this reminds me how we do the upgrade

```
   sequential:
   - install.upgrade:
       mon.a:
       mon.b:
   - ceph.restart: [mon.a, mon.b, mon.c, mds.a, osd.0, osd.1, osd.2, osd.3]
```

so i thought we didn't want to stop the service until upgrade is finished. but yeah. i can drop that option.

Comment (tchaikov) [inline debian/rules]: > i think we want --no-restart-on-upgrade

okay. if we expect our user to start the upgraded service manually.

**Comment by tchaikov** [comment]
changelog

- s/--restart-after-upgrade/--no-restart-on-upgrade/

**Comment by liewegas** [comment]
lgtm!  @javacruft ?

**Comment by tchaikov** [comment]
@javacruft ping?

**Comment by jmunhoz** [comment]
Works on Jessie amd64! btw the commit description uses the old Fixes: # format instead of the new URL format

**Comment by tchaikov** [comment]
changelog

- update the commit message to fix the "Fixed" label

**Comment by ramsberg** [comment]
Hi. Sorry for a lay-man question: When can this fix be expected to be implemented in deb packages for ceph hammer for armhf/jessie (raspberry Pi) ?

**Comment by tchaikov** [comment]
@ramsberge it's a known issue, see http://tracker.ceph.com/issues/11344. but i see there are still some open questions documented in that ticket not yet answered.

**Comment by ramsberg** [comment]
Thanks Kefu. 
It is a little troubling though, that this discussion is over one year old with no directly constructive updates added. 
Is that normal? 
/Olle 
On May 1, 2016 16:27, Kefu Chai <notifications@github.com> wrote:@ramsberge it's a known issue, see http://tracker.ceph.com/issues/11344. but i see there are still some open questions documented in that ticket not yet answered.

**Comment by liewegas** [comment]
The systemd issues are all settled--I closed that bug.

For hammer, we don't have systemd files backported, and don't plan to.  If you want systemd, upgrade to jewel.

Note that we are only doing armhf jessie builds on jewel and later, and only test builds currently--no release ones.

**Comment by ramsberg** [comment]
Ahh! Got it! Will dig into getting jewel running on raspbian/armhf/jessie!
Thanks!

**Pull request merged**